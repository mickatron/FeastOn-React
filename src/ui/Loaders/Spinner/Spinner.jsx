import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

import classNameType from 'js/shared/proptypes/className';


const Spinner = function({ children, className }) {
  return (
    <div className={classnames(className,'c-Spinner')}>

      { children ? 
        <span className="c-Spinner__text">{children}</span> 
        : 
        null 
      }
    </div>
  );
};

Spinner.propTypes = {
  children: PropTypes.node,
  classNameType,
};

export default Spinner;