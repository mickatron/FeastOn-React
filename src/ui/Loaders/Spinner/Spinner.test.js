import { shallow } from 'enzyme';
import React from 'react';
import Spinner from './Spinner';

describe('Spinner', () => {

  let component = null;
  describe('renders', () => {
    it('Spinner with children', () => {
      component = shallow(
        <Spinner
          className="test"
        >
          <p>Some Spinner test text.</p>
        </Spinner>
      );
      expect(component).toMatchSnapshot();
    });
    
    it('Spinner without children', () => {
      component = shallow(
        <Spinner
          className="test"
        />
      );
      expect(component).toMatchSnapshot();
    });

  });

});