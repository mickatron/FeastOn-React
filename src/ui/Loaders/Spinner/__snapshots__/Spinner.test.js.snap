// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders Spinner with children 1`] = `
<div
  className="test c-Spinner"
>
  <span
    className="c-Spinner__text"
  >
    <p>
      Some Spinner test text.
    </p>
  </span>
</div>
`;

exports[`Spinner renders Spinner without children 1`] = `
<div
  className="test c-Spinner"
/>
`;
